Expensify/App
# Enable again all address fields and allow the user to manually fill them / Move form errors to component state

Sorry for the long PR

Pullerbear: nickmurray47

### Details


- Added `state`, `city` and `zipCode` fields back in the following steps: "Company Step", "Personal information (requestor)" and "Additional information (beneficial owners)"
- When we select a result from the address autocomplete, we use the information that is available. If some part is missing (i.e. city), we just ignore that.
- We only replace the `addressStreet` if the `street name + street number` coming from the google places result is longer than what the user typed.
- Removed `googlePlacesRef` as I don't see the point of having that anymore. We can just pass the value down that is coming from the form as `textInputProps`
- Introduced `skippedFirstOnChangeText` state to skip the first time the library calls our text input's `onChangeText` callback. I'm not sure why (doesn't make sense to me), the library calls `onChangeText` with an empty string `""` to clear the value. Clearing the value in our case is not something we want when we are returning to a step that we already filled.
- Deleted unused validation method for Google places address components
- `IdentityForm` was reading address inputs using keys `city`, `zipCode` and `state`, but it was calling `onFieldChange` using `addressCity`, `addressZipCode` and `addressState`. Made it more consistent and changed it to always use `addressCity`, `addressZipCode` and `addressState`.
- Converted `AddressSearch` to class component
- Moved the form validation errors from Onyx to the form component state because of [problem explained here](https://github.com/Expensify/App/pull/6200#issuecomment-963638250)

### Fixed Issues

$ https://github.com/Expensify/App/issues/6309
$ https://github.com/Expensify/Expensify/issues/181598
$ https://github.com/Expensify/App/issues/5841

### Tests / QA steps

1. Create NewDot account, verify it.
2. Create workspace and start the add VBA flow
3. In "Company information" step:
    1. You should see the inputs: `Company address`, `City`, `Zip Code`, and `State`:
        <img src="https://user-images.githubusercontent.com/87341702/141878056-fa146760-ddd5-4717-9346-d3a21ae836aa.png" width="368">
    2. Click "Save & Continue", they should get inline errors
        <img src="https://user-images.githubusercontent.com/87341702/141878184-5bead448-7690-4dde-ba60-b49df7084038.png" width="368">
    3. Typing in any of the field should clear the error.
    4. Type and address in `Company address` and select a result. The other address fields (`City`, `Zip Code`, and `State`) should get updated with the information from the selected result.
    5. Focus on the `Company address` field and type `#12334` at the end (adding the apt number).
    6. Click "Save & Continue", you should be able to proceed to the next step.
4. Do the same steps (sub-steps in step (3)) for "Personal information"
5. Do the same steps (sub-steps in step (3)) for "Additional information" adding beneficial owners

### Tested On


### Screenshots


#### Web


#### Mobile Web


#### Desktop


#### iOS


#### Android
